intake/__init__.py

```diff
@@ -24,6 +26,9 @@ def make_open_functions():
     """From the current state of ``registry``, create open_* functions"""
     for plugin_name, plugin in registry.items():
         func_name = 'open_' + plugin_name
+        if not isidentifier(func_name):
+            # primitive name normalization
+            func_name = re.sub('[-=~^&|@+]', '_', func_name)
```

Comment:
Can we also emit a warning that we've done this renaming?  Plugins should generally use identifier-friendly names and not rely on a specific name normalization scheme.

Reply:
OK, I'm happy to do the check in the base class.